eng/native/configurecompiler.cmake

```diff
@@ -448,6 +452,7 @@ if (CLR_CMAKE_HOST_UNIX)
     add_compile_options(-Wno-incompatible-ms-struct)
 
     add_compile_options(-Wno-reserved-identifier)
+    add_compile_options(-Wno-sign-conversion)
```

Comment:
why do you need to disable this warning?

Reply:
Building for `ppc64le` with clang-15 gives out `Wsign-conversion` errors. I can reproduce the error if you require the exact error message.

Reply:
Did you observe those warnings outside of `src/mono`? If not, please move this suppression to one of the `CMakeLists.txt` file under src/mono. Note that we are building coreclr with clang-15 in arm64 CI leg.

Reply:
Here is the exact error:
```
/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/native/libs/System.Native/pal_io.c(1270,36): error GBBD8A67C: implicit conversion changes signedness: 'unsigned long' to 'int' [-Wsign-conversion] [/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/native/libs/build-native.proj]
/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/native/libs/System.Native/pal_io.c(1270,36): error GBBD8A67C: implicit conversion changes signedness: 'unsigned long' to 'int' [-Wsign-conversion] [/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/native/libs/build-native.proj]
```

Reply:
`(int)FICLONE` at:
https://github.com/dotnet/runtime/blob/37191cd6bcd0d7062b1c5b84fd857f8e261c25e9/src/native/libs/System.Native/pal_io.c#L1270

Reply:
That did it, thanks! Pushed without no-sign-conversion.